### PR TITLE
Add delete Panel Survey

### PR DESCRIPTION
**Pull request by ftarulla**
Related #2029 

- Add Delete Panel Survey feature.
- The card of a Panel Survey now have 3 options:
   - Move
   - Delete wave (delete the wave shown - ie. the last wave)
   - Delete Panel Survey

### CODE REVIEW

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 1]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
 
   render() {
-    const { survey, respondentsStats, t, panelSurveyId } = this.props
-    const { panelSurvey } = survey
+    const { panelSurvey, t, dispatch } = this.props
+    const survey = this.latestWave()
+
+    let actions = []
+    if (this.movable()) actions.push({ name: t('Move to'), func: this.askMovePanelSurvey })
+    if (this.deletable()) actions.push({ name: t('Delete wave'), func: this.askDeleteSurvey })
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:208]: Maybe `askDeleteWave` or even more explicit `askDeletePanelSurveyWave`?

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 2]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
+
+  actionable() {
+    return this.deletable() || this.movable()
+  }
+
+  render() {
+    const { survey, t, dispatch } = this.props
+
+    let actions = []
+    if (this.movable()) actions.push({ name: t('Move to'), func: this.askMoveSurvey })
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:94]: I couldn't move a survey anymore, so I changed the callback back to `moveSurvey()` (which exists) and it worked!

```suggestion
    if (this.movable()) actions.push({ name: t('Move to'), func: this.moveSurvey })
```

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 3]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
-        </div>
-      </Card>
-    </div>
+ActionMenu.propTypes = {
+  actions: PropTypes.array
+}
+
+class ConfirmationCheckModal extends Component<Props, State> {
+  static propTypes = {
+    modalId: PropTypes.String,
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:317]: Typo:
```suggestion
    modalId: PropTypes.string,
```

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 4]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
 
     return (
-      <div className='col s12 m6 l4'>
-        {
-          showAsPanelSurveyCard
-          ? <div className='panel-survey-card-0'>
-            <div className='panel-survey-card-1'>
-              <div className='panel-survey-card-2'>
-                { surveyCard }
+      <div style={{ 'text-align': 'center' }}>
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:358]: React complains that the `text-align` property doesn't exist:
```suggestion
      <div style={{ textAlign: 'center' }}>
```

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 5]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
+        <Modal card id={modalId} ref='modal'>
+          <div className='modal-content'>
+            <div className='card-title header' style={{'margin-bottom': '48px'}}>
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:361]: Same here, React complains that the `margin-bottom` property doesn't exist:

```suggestion
            <div className='card-title header' style={{marginBottom: '48px'}}>
```

File: web/static/js/components/surveys/SurveyCard.jsx [hunk 6]
```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
...
-            </Link>
-          </div>
-        </div>
-      </Card>
-    </div>
+ActionMenu.propTypes = {
+  actions: PropTypes.array
+}
+
+class ConfirmationCheckModal extends Component<Props, State> {
```
Comment (ysbaddaden) [inline web/static/js/components/surveys/SurveyCard.jsx:315]: I'm not sure the naming is explicit on its differences with the regular ConfirmationModal component.

We're display a "warning" and need to tick a checkbox to "consent" with the action, so maybe something `WarningModal` or `ExplicitConfirmationModal` or something else?

**Review by ysbaddaden — COMMENTED** [review]
I used https://github.com/instedd/surveda/compare/feature/1981-move-panelsurvey-folder-3...feature/2029-delete-panelsurvey to review this pull request.

It works great!

I only noticed a few typos which I outlined below. One led to not being able to move a survey anymore, and is easily fixed. I'm also proposing to change a few namings to help clarify the intents.

But that's about it. Good job!

File: web/static/js/components/surveys/SurveyCard.jsx (hunk 1 above)
Comment (ftarulla) [reply web/static/js/components/surveys/SurveyCard.jsx:208]: yeah! Totally agree!!

File: web/static/js/components/surveys/SurveyCard.jsx (hunk 6 above)
Comment (ftarulla) [reply web/static/js/components/surveys/SurveyCard.jsx:315]: Both `ConfirmationModal` and `ConfirmationCheckModal` are warnings. The only difference is the extra step of clicking a check input to enable the confirmation button. So maybe we can call it `TwoStepsConfirmationModal`?

File: web/static/js/components/surveys/SurveyCard.jsx (hunk 2 above)
Comment (ftarulla) [reply web/static/js/components/surveys/SurveyCard.jsx:94]: yeah! A replace error 🙈 
I'm rewriting this to a `ask<action>` and `confirm<action>` pattern (like in `PanelSurveyCard`)

File: web/static/js/components/surveys/SurveyCard.jsx (hunk 6 above)
Comment (ysbaddaden) [reply web/static/js/components/surveys/SurveyCard.jsx]: Oh `TwoStepsConfirmationModal` is nice!

**Review by ysbaddaden — APPROVED** [review]

**Comment by ysbaddaden** [comment]
Rebased on `main` now that #2037 is merged. Waiting for CI before then LGTM too!

**Pull request merged**